lib/services/services_linux.c

```diff
@@ -764,7 +764,18 @@ resources_os_list_lsb_agents(void)
 GList *
 resources_os_list_ocf_providers(void)
 {
-    return get_directory_list(OCF_ROOT_DIR "/resource.d", FALSE, TRUE);
+    GList *gIter = NULL;
+    GList *providers = NULL;
+
+    providers = get_directory_list(OCF_ROOT_DIR "/resource.d", FALSE, TRUE);
+    for (gIter = providers; gIter != NULL; gIter = gIter->next) {
+        if (safe_str_eq(gIter->data, LRMD_ISOLATION_PROVIDER)) {
+            providers = g_list_remove_link(providers, gIter);
+            g_list_free_1(gIter);
```

Comment:
what? How is this not safe? Since when did break behave like this? By this logic the following code should result in printing 11 instead of 10.

int i;
for ( i=1; i <=999999; i = i +1 ){
if (i == 10) {
break;
}
}

        printf("I IS %d\n", i);